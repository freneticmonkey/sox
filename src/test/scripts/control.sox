var true_val = true;
var false_val = false;

print "Testing If Statements";

if (true_val) {
    print "true_val is true";
} else {
    print "true_val is NOT true. Error!";
}

if (true_val == false) {
    print "true_val == false. Error!";
} else {
    print "true_val != false";
}

if (true_val and false_val) {
    print "true_val and false_val. Error!";
} else {
    print "true_val and false_val. Success";
}

if (true_val and true_val) {
    print "true_val and true_val. Success!";
} else {
    print "true_val and true_val. Error";
}

if (true_val or false_val) {
    print "true_val or false_val. Success";
} else {
    print "true_val or false_val. Error!";
}

if (false_val or false_val) {
    print "false_val or false_val. Error";
} else {
    print "false_val or false_val. Success!";
}


print "Testing While";
var initial = 0;

while (initial < 10) {
    initial = initial + 1;
    print initial;
}

print "Testing For";
for (var i = 0; i < 10; i = i + 1) {
    print i;
}