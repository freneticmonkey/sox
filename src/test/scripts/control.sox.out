Testing If Statements
true_val is true
true_val != false
true_val and false_val. Success
true_val and true_val. Success!
true_val or false_val. Success
false_val or false_val. Success!
true_val is true. Success
else Success.
